src/main/java/com/rtb/projectmanagementtool/user/UserProfileServlet.java

```diff
@@ -57,37 +57,47 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     // get user with ID
     UserData user = userController.getUserByID(userID);
 
+    // simple way to have majors without []
+    String userMajorsString = "";
+    for (String major : user.getUserMajors()) {
+      if (userMajorsString.equals("")) {
+        userMajorsString = major;
+      } else {
+        userMajorsString = userMajorsString + ", " + major;
+      }
+    }
+
     // get this user's tasks
     ArrayList<TaskData> tasks = taskController.getTasksByUserID(userID);
 
-    // this should only be sent load user looking @ own profile
-
-    // build list of private comments and colors
+    // build list of private comments
     ArrayList<PrivateCommentData> privateComments = new ArrayList<>();
 
-    // build private comments
+    // get private comments
     privateComments = privateCommentController.getPrivateCommentsForUser(userID);
 
-    // build map of task to pcomment
+    // build map of taskID to pcomment
     Map<Long, PrivateCommentData> privateCommentsMap = new HashMap<Long, PrivateCommentData>();
 
-    // preset each comment for each task empty
-    for (TaskData task : tasks) {
-      privateCommentsMap.put(
-          task.getTaskID(), new PrivateCommentData(task.getTaskID(), userID, ""));
+    // preset each comment if every task is empty
```

Comment:
I think this is the right idea, creating a blank comment object if the task does not have a comment yet so you won't have to deal with nulls later on.  But, right now you're only doing the pre-fill if all comments for all tasks are missing.  For example if they have 4 tasks with no comments, initially this will work great to pre-fill with black comments.  But, once they add a comment for one of those 4 tasks, then the next time they go to this page only the comments for that 1 task will be filled (with legit data this time), and then you're back dealing with nulls on your jsp

Reply:
I think my most recent commit will fix this issue